File: BringMeHome/BringMeHome.Services/Database/AdoptionApplication.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using static BringMeHome.Services.Database.Adopters;

namespace BringMeHome.Services.Database
{
    public class AdoptionApplication
    {
        [Key]
        public int ApplicationID { get; set; }

        public int UserID { get; set; }

        public int AnimalID { get; set; }

        [DataType(DataType.Date)]
        public DateTime ApplicationDate { get; set; }

        public int StatusID { get; set; }

        public int? ReviewedByStaffID { get; set; }

        [DataType(DataType.Date)]
        public DateTime? ReviewDate { get; set; }

        [StringLength(1000)]
        public string? Notes { get; set; }

        public string? LivingSituation { get; set; }

        public string? IsAnimalAllowed { get; set; }

        public int? ReasonID { get; set; } 

        // Navigation properties
        [ForeignKey("UserID")]
        public virtual User User { get; set; }

        [ForeignKey("AnimalID")]
        public virtual Animal Animal { get; set; }

        [ForeignKey("StatusID")]
        public virtual ApplicationStatus Status { get; set; }

        [ForeignKey("ReviewedByStaffID")]
        public virtual Staff ReviewedBy { get; set; }

        [ForeignKey("ReasonID")]
        public virtual Reason? Reason { get; set; }

    }
}

File: bringmehomeUI/bringmehome_android/lib/models/adoption_application.dart
import 'package:learning_app/models/user.dart';
import 'package:learning_app/models/animal.dart';
import 'package:learning_app/models/application_status.dart';

class AdoptionApplication {
  final int? applicationID;
  final int userID;
  final int animalID;
  final DateTime? applicationDate; 
  final int statusID; 
  final String? notes; 
  final String? livingSituation;
  final String? isAnimalAllowed; 
  final int? reasonId;

  final User? user; 
  final Animal? animal;
  final ApplicationStatus? status;
  final String? statusName;
  final User? reviewedBy; 

  AdoptionApplication.create({
    required this.userID,
    required this.animalID,
    this.notes,
    this.livingSituation,
    this.isAnimalAllowed,
    this.reasonId,
    this.applicationID,
    this.applicationDate,
    this.statusID = 1, 
    this.reviewedBy,
    this.status,
    this.statusName,
    this.user, 
    this.animal,
  });


  factory AdoptionApplication.fromJson(Map<String, dynamic> json) {
    final userJson = json['user'];
    final animalJson = json['animal'];
    final statusJson = json['status'];
    final reviewedByJson = json['reviewedBy']; 

    return AdoptionApplication(
      applicationID: json['applicationID'] as int, 
      userID: json['userID'] as int, 
      animalID: json['animalID'] as int,
      applicationDate: json['applicationDate'] != null
          ? DateTime.parse(json['applicationDate'] as String)
          : null,
      statusID: json['statusID'] as int,
      notes: json['notes'] as String?,
      livingSituation: json['livingSituation'] as String?,
      isAnimalAllowed: json['isAnimalAllowed'] as String?,
      statusName: json['statusName'] as String?,
      reasonId:json['reasonId'] as int?,
      user: userJson != null ? User.fromJson(userJson as Map<String, dynamic>) : null,
      animal: animalJson != null ? Animal.fromJson(animalJson as Map<String, dynamic>) : null,
      status: statusJson != null ? ApplicationStatus.fromJson(statusJson as Map<String, dynamic>) : null,
       reviewedBy: reviewedByJson != null ? User.fromJson(reviewedByJson as Map<String, dynamic>) : null, 
    );
  }

  AdoptionApplication({
      this.applicationID,
      required this.userID,
      required this.animalID,
      this.applicationDate,
      required this.statusID,
      this.notes,
      this.livingSituation,
      this.isAnimalAllowed,
      this.reasonId,
      this.user,
      this.statusName,
      this.animal,
      this.status,
      this.reviewedBy,
  });

  Map<String, dynamic> toJson() {
    return {
      if(applicationID != null) 'applicationID': applicationID,
      'userID': userID,
      'animalID': animalID,
      if(applicationDate != null) 'applicationDate': applicationDate!.toIso8601String(), 
      'statusID': statusID,
      if(statusName != null) 'statusName': statusName,      
      'notes': notes,
      'livingSituation': livingSituation,
      'isAnimalAllowed': isAnimalAllowed,
      'reasonId': reasonId, 
    };
  }
}